c++/src/capnp/reconnect.c++

```diff
@@ -27,23 +27,33 @@ namespace {
 
 class ReconnectHook final: public ClientHook, public kj::Refcounted {
 public:
-  ReconnectHook(kj::Function<Capability::Client()> connectParam)
+  ReconnectHook(kj::Function<Capability::Client()> connectParam, bool lazy = false)
       : connect(kj::mv(connectParam)),
-        current(ClientHook::from(connect())) {}
+        current(lazy ? kj::Maybe<kj::Own<ClientHook>>() : ClientHook::from(connect())) {}
 
   Request<AnyPointer, AnyPointer> newCall(
       uint64_t interfaceId, uint16_t methodId, kj::Maybe<MessageSize> sizeHint) override {
-    auto result = current->newCall(interfaceId, methodId, sizeHint);
-    AnyPointer::Builder builder = result;
-    auto hook = kj::heap<RequestImpl>(kj::addRef(*this), RequestHook::from(kj::mv(result)));
-    return { builder, kj::mv(hook) };
+    KJ_IF_MAYBE(c, current) {
+      auto result = (*c)->newCall(interfaceId, methodId, sizeHint);
+      AnyPointer::Builder builder = result;
+      auto hook = kj::heap<RequestImpl>(kj::addRef(*this), RequestHook::from(kj::mv(result)));
+      return { builder, kj::mv(hook) };
+    } else {
+      current = ClientHook::from(connect());
+      return newCall(interfaceId, methodId, sizeHint);
```

Comment:
This recursion strikes me as a little odd. What about having a helper method like:

    ClientHook& getCurrent() {
      KJ_IF_MAYBE(c, current) {
        return **c;
      } else {
        return current.emplace(ClientHook::from(connect()));
      }
    }

Then call that in the other methods.

Reply:
Agreed, done.